ee/experimental/incident_utils.py

```diff
@@ -343,10 +385,12 @@ def generate_incident_summary(incident: Incident, use_n_alerts_for_summary: int
     try: 
         client = OpenAI(api_key=os.environ["OPENAI_API_KEY"])
         
+        incident = get_incident_by_id(incident.tenant_id, incident.id)
         prompt_addition = ''
         if incident.user_summary:
             prompt_addition = f'When generating, you must rely on the summary provided by human: {incident.user_summary}'
         
+        alert_number = len(incident.alerts)
```

Comment:
to prevent getting all alerts from the database:

```suggestion
        alert_number = incident.alerts_count
```